inst/stan/epinowcast.stan

```diff
@@ -8,16 +8,19 @@ functions {
 
 data {
   // Indexes and lookups
+  int n; // total observations
   int t; // time range over which data is available 
   int s; // number of snapshots there are
   int g; // number of data groups
   int st[s]; // when in this time snapshots are from
   int ts[t, g]; // snapshot related  to time and group
   int sl[s]; // how many days of reported data does each snapshot have
+  int csl[s]; // cumulative version of the above
```

Comment:
Out of interest, have you also thought about the alternative strategy of postponing the flattening to the model and doing it in `generated_data`? It would be slightly less efficient, because the flattening is done once by each chain. The advantage could be to have a more concise and intuitive data list on the other hand... I am just interested in your thoughts, quite impartial regarding this...

Reply:
So I did consider this but I have been trying to keep as much code in R vs in stan as possible so that it can be unit tested (obviously not at the moment) which was the main driver behind this approach. The other reason is that if we transition the back-end or go back-end agnostic in some far future having more code in R vs in stan is likely going to be helpful.